### PR TITLE
sdk 50 v2

### PR DESCRIPTION
**Pull request by faddat**
This PR updates interchaintest to sdk v50.

### CODE REVIEW

**Comment by faddat** [comment]
ohhh great point @catShaark -- I'd neglected that step, thank you!

**Comment by faddat** [comment]
@catShaark -- I was able to stop this annoying linter issue by explicitly specifying that the linter action should:

* install the latest go
* use the latest golangci-lint

File: go.mod
Comment (boojamya) [inline go.mod:266]: I was told by IBC-Go team that here is an alpha version that uses sdk 50:
https://github.com/cosmos/ibc-go/tree/upgrade-sdk-v0.50-alpha1

We should use this instead.

Comment (faddat) [reply go.mod:266]: we should not.  

the alpha version there is not up to date with the sdk, and there have been breaking changes between the two.  

I agree that it is not good practice to have a replace statement like this one, so I'll see about getting a new release cut asap from this:

* https://github.com/cosmos/ibc-go/pull/4392

I'll come back tomorrow, I do reckon that we can get the release cut in the next 24h.

File: .github/workflows/lint.yml
Comment (faddat) [reply .github/workflows/lint.yml:18]: makes it so it doesn't periodically explode on ya.

File: chain/cosmos/chain_node.go
Comment (faddat) [reply chain/cosmos/chain_node.go:420]: I'd need to defer to @catShaark on this.  I am not sure, but thanks for the review and bringing this up!

**Comment by faddat** [comment]
quick update -- we've gotten this PR:

* https://github.com/cosmos/ibc-go/pull/4391

ready to merge and it.... more or less means that `main` in ibc-go is ready to go for v50.  So, most likely we will release from there instead.

File: chain/cosmos/chain_node.go
Comment (catShaark) [reply chain/cosmos/chain_node.go:420]: thanks @misko9, I just add the code u request.

**Comment by boojamya** [comment]
@faddat made some progress here. I should have just pushed directly to this branch. But made a PR here:
https://github.com/notional-labs/interchaintest/pull/7

Maybe don't squash  and just merge so we can keep track of the commits?

I removed `wasm`, `polkadot` and `hyperspace` as those are not supported by sdk 50 yet. Was running into go mod issues. (cc @misko9 )

I also updated the boundary test. It seems like the relayer is having trouble creating clients on chains running sdk 50.
Will pass this along to the team.

**Comment by boojamya** [comment]
We are really close. @charleenfei, any chance you know why the CI doesn't like the hermes config?

https://github.com/strangelove-ventures/interchaintest/actions/runs/6042953523/job/16399102419?pr=716#step:5:55

`failed to configure relayer r for chain gaia-9: exit code 1:  The Hermes configuration file at path '/home/hermes/.hermes/config.toml' is invalid, reason: config error: path error: /home/hermes/.hermes/config.toml`

**Comment by faddat** [comment]
path error sort of looks like it cannot find the file, but I am not sure that is the actual issue here. 

It is also possible that the configuration schema has changed, if you're upgrading hermes from 1.5* to 1.6*

**Comment by charleenfei** [comment]
Yea, we solved this on our side. Basically since 1.4.1, Hermes is compiled
with Ubuntu 23 which has a default first User uid:gid of 1001:1001 instead
of 1000:1000, we need to update tjat for the Hermes config.

**Comment by boojamya** [comment]
Module bumped
from: `github.com/strangelove-ventures/interchaintest/v7`
to: `github.com/strangelove-ventures/interchaintest/v8`

Right before I merge, I will branch off off a `v7`. 
`v8` will be the new main.

**Comment by boojamya** [comment]
Note: since this is from a fork, I think it will be easiest to update `local-interchain` once this is merged. 

Reason: I can't properly import the right commit of `interchaintest` in the go mod.

cc: @Reecepbcups

**Review by boojamya — APPROVED** [review]
This looks like it is good to go! Thanks all!

Boundary tests (which runs IBC conformance tests) are passing on a simd image built with that alpha version of ibc-go v8.

Will update Readme in follow up PR.

**Comment by Reecepbcups** [comment]
Ahh, yea this requires an odd go.mod quirk to resolve. Will fix after merge, does not seem I can push up to this PR

File: examples/cosmos/sdk_boundary_test.go
Comment (charleenfei) [inline examples/cosmos/sdk_boundary_test.go:45]: qq: do you guys have an eta for when the golang relayer going to be updated with the changes in @colin-axner's image?

**Comment by colin-axner** [comment]
We have a test which uses [`UnsafeResetAll()`](https://github.com/strangelove-ventures/interchaintest/blob/main/chain/cosmos/chain_node.go#L1079) and it needs to be updated like so:
```diff
diff --git a/chain/cosmos/chain_node.go b/chain/cosmos/chain_node.go
index 623f37c..02bcb50 100644
--- a/chain/cosmos/chain_node.go
+++ b/chain/cosmos/chain_node.go
@@ -1055,7 +1055,7 @@ func (tn *ChainNode) UnsafeResetAll(ctx context.Context) error {
        tn.lock.Lock()
        defer tn.lock.Unlock()
 
-       _, _, err := tn.ExecBin(ctx, "unsafe-reset-all")
+       _, _, err := tn.ExecBin(ctx, "comet", "unsafe-reset-all")
        return err
 }
```

Is there a way we can handle backwards compatibility here?

File: examples/cosmos/sdk_boundary_test.go
Comment (jtieri) [reply examples/cosmos/sdk_boundary_test.go:45]: just approved colin's pr! should be merged once CI passes

**Comment by Reecepbcups** [comment]
@colin-axner Yes, we do need to handle backwards compatibility here.

After this PR is merged, we can use the [UseNewGenesisCommand](https://github.com/strangelove-ventures/interchaintest/blob/3f614f126707bdaf673d7f83a3979f210e9a4fab/local-interchain/cmd/local-ic/new_chain.go#L72) bool to toggle the cmd format

**Pull request merged**
